### PR TITLE
feat: Add txIndex for BlockPublisher handlers

### LINKED ISSUE #100: Add txIndex for BlockPublisher handlers

### PR DESCRIPTION
**Pull request by fqutishat**
closes #100

Signed-off-by: Firas Qutishat <firas.qutishat@securekey.com>

### CODE REVIEW

File: extensions/gossip/api/gossipapi.go [hunk 1]
```diff
@@ -42,3 +42,11 @@ type BlockPublisher interface {
 	// Publish traverses the block and invokes all applicable handlers
 	Publish(block *cb.Block)
 }
+
+// TxnInfo contain txn info
+type TxnInfo struct {
+	BlockNum  uint64
+	ChannelID string
+	TxID      string
```
Comment (troyronda) [inline extensions/gossip/api/gossipapi.go]: Stutter?

Comment (troyronda) [reply extensions/gossip/api/gossipapi.go]: (also I see multiple representations of transaction)

File: extensions/gossip/api/gossipapi.go [hunk 2]
```diff
@@ -42,3 +42,11 @@ type BlockPublisher interface {
 	// Publish traverses the block and invokes all applicable handlers
 	Publish(block *cb.Block)
 }
+
+// TxnInfo contain txn info
```
Comment (troyronda) [inline extensions/gossip/api/gossipapi.go]: Does it really? :)

File: extensions/gossip/api/gossipapi.go (hunk 1 above)
Comment (fqutishat) [reply extensions/gossip/api/gossipapi.go]: done

File: extensions/gossip/api/gossipapi.go (hunk 2 above)
Comment (fqutishat) [reply extensions/gossip/api/gossipapi.go]: done

File: extensions/gossip/api/gossipapi.go [hunk 3]
```diff
@@ -42,3 +42,11 @@ type BlockPublisher interface {
 	// Publish traverses the block and invokes all applicable handlers
 	Publish(block *cb.Block)
 }
+
+// TxMetadata contain txn metadata
+type TxMetadata struct {
+	BlockNum  uint64
+	ChannelID string
+	ID        string
+	Num       uint64
```
Comment (troyronda) [inline extensions/gossip/api/gossipapi.go]: This structure could use a comment, as BlockNum vs Num will be confusing.

These together are referring to Height (https://github.com/hyperledger/fabric/blob/bbd0324b247e9e9d24ff445c7234225aaf552f4d/core/ledger/kvledger/txmgmt/version/version.go#L26)

**Pull request merged**